org.idempiere.test/src/org/idempiere/test/base/InOutTest.java

```diff
@@ -0,0 +1,434 @@
+		MConversionRate crUsdSpot = createConversionRate(australianDollar.getC_Currency_ID(), usd.getC_Currency_ID(), Spot_ConversionType_ID, currentDate, audToUsdSpot);
+		
+		MCurrency euro = MCurrency.get("EUR"); // EUR
+		BigDecimal audToEuroCompany = new BigDecimal(0.746234);
+		MConversionRate crEurCompany = createConversionRate(australianDollar.getC_Currency_ID(), euro.getC_Currency_ID(), Company_ConversionType_ID, currentDate, audToEuroCompany);
+		BigDecimal audToEuroSpot = new BigDecimal(0.64);
+		MConversionRate crEurSpot = createConversionRate(australianDollar.getC_Currency_ID(), euro.getC_Currency_ID(), Spot_ConversionType_ID, currentDate, audToEuroSpot);
+
+		try {
+			MOrder order = createPurchaseOrder(bpartner, currentDate, priceList.getM_PriceList_ID(), Company_ConversionType_ID);			
+			BigDecimal qtyOrdered = new BigDecimal(500);
+			MOrderLine orderLine = createPurchaseOrderLine(order, 10, product, qtyOrdered, priceInAud);
+			completeDocument(order);
+			
+			MInOut receipt = createMMReceipt(order, currentDate);			
+			BigDecimal qtyDelivered = new BigDecimal(500);
+			MInOutLine receiptLine = createMMReceiptLine(receipt, orderLine, qtyDelivered);
+			completeDocument(receipt);
+			postDocument(receipt);
+			
+			MAcctSchema[] ass = MAcctSchema.getClientAcctSchema(Env.getCtx(), Env.getAD_Client_ID(Env.getCtx()));
+			for (MAcctSchema as : ass) {
+				BigDecimal rate = Env.ZERO;
+				if (as.getC_Currency_ID() == usd.getC_Currency_ID())
+					rate = audToUsdCompany;
+				else if (as.getC_Currency_ID() == euro.getC_Currency_ID())
+					rate = audToEuroCompany;
+					
+				Doc doc = DocManager.getDocument(as, MInOut.Table_ID, receipt.get_ID(), getTrxName());
+				doc.setC_BPartner_ID(receipt.getC_BPartner_ID());
+				MAccount acctNIR = doc.getAccount(Doc.ACCTTYPE_NotInvoicedReceipts, as);
+				
+				String whereClause = MFactAcct.COLUMNNAME_AD_Table_ID + "=" + MInOut.Table_ID 
+						+ " AND " + MFactAcct.COLUMNNAME_Record_ID + "=" + receipt.get_ID()
+						+ " AND " + MFactAcct.COLUMNNAME_C_AcctSchema_ID + "=" + as.getC_AcctSchema_ID();
+				int[] ids = MFactAcct.getAllIDs(MFactAcct.Table_Name, whereClause, getTrxName());
+				for (int id : ids) {
+					MFactAcct fa = new MFactAcct(Env.getCtx(), id, getTrxName());
+					if (acctNIR.getAccount_ID() == fa.getAccount_ID()) {
+						if (receiptLine.get_ID() == fa.getLine_ID()) {
+							BigDecimal acctSource = orderLine.getPriceActual().multiply(receiptLine.getMovementQty())
+									.setScale(as.getC_Currency().getStdPrecision(), RoundingMode.HALF_UP);
+							BigDecimal acctAmount = acctSource.multiply(rate)
+									.setScale(as.getC_Currency().getStdPrecision(), RoundingMode.HALF_UP);
+							assertTrue(fa.getAmtSourceCr().compareTo(acctSource) == 0, fa.getAmtSourceCr().toPlainString() + " != " + acctSource.toPlainString());
+							assertTrue(fa.getAmtAcctCr().compareTo(acctAmount) == 0, fa.getAmtAcctCr().toPlainString() + " != " + acctAmount.toPlainString());							
+						}
```

Comment:
should have assert to check the accounted amount is using the expected conversion rate